src/components/FileUploaderV2/FileUploaderV2.js

```diff
@@ -0,0 +1,240 @@
+    labelText: 'Add file',
+    buttonKind: 'primary',
+    multiple: false,
+    onChange: () => {},
+    onClick: () => {},
+    accept: [],
+  };
+  uid = this.props.id || uid();
+
+  render() {
+    const {
+      className,
+      multiple,
+      role,
+      tabIndex,
+      buttonKind,
+      accept,
+      name,
+      onChange,
+      labelText,
+      ...other
+    } = this.props;
+    const classes = classNames({
+      'bx--file': true,
+      [className]: className,
+    });
+
+    return (
+      <div
+        role="button"
+        tabIndex="0"
+        className={classes}
+        onKeyDown={evt => {
+          if (evt.which === 13 || evt.which === 32) {
+            this.input.click();
+          }
+        }}>
+        <label
+          className={`bx--btn bx--btn--${buttonKind}`}
+          tabIndex={tabIndex}
+          htmlFor={this.uid}
+          role={role}
+          {...other}>
+          {labelText}
+        </label>
+        <input
+          className="bx--visually-hidden"
+          ref={input => (this.input = input)}
+          id={this.uid}
+          type="file"
+          multiple={multiple}
+          accept={accept}
+          name={name}
+          onChange={onChange}
+          onClick={evt => {
+            evt.target.value = null;
+          }}
+        />
+      </div>
+    );
+  }
+}
+
+export function FilenameV2({
+  iconDescription,
+  onKeyDown,
+  status,
+  style,
+  tabIndex,
+  ...other
+}) {
+  switch (status) {
+    case 'uploading':
+      return (
+        <div
+          className="bx--loading"
+          style={{ ...style }}
+          tabIndex={tabIndex}
+          onKeyDown={onKeyDown}
+          role="button"
+          {...other}>
+          <svg className="bx--loading__svg" viewBox="-42 -42 84 84">
+            <circle cx="0" cy="0" r="37.5" />
+          </svg>
+        </div>
+      );
+    case 'edit':
+      return (
+        <Icon
+          description={iconDescription}
+          className="bx--file-close"
+          icon={iconCloseSolid}
+          style={style}
+          tabIndex={tabIndex}
+          onKeyDown={onKeyDown}
+          role="button"
+          {...other}
+        />
+      );
+    case 'complete':
+      return (
+        <Icon
+          description={iconDescription}
+          className="bx--file-complete"
+          icon={iconCheckmarkSolid}
+          style={style}
+          tabIndex={tabIndex}
+          onKeyDown={onKeyDown}
+          role="button"
+          {...other}
+        />
+      );
+    default:
+      return null;
+  }
+}
+FilenameV2.propTypes = {
+  onKeyDown: PropTypes.func,
+  style: PropTypes.object,
+  status: PropTypes.oneOf(['edit', 'complete', 'uploading']),
+  tabIndex: PropTypes.number,
+};
+FilenameV2.defaultProps = {
+  onKeyDown: () => {},
+  status: 'uploading',
+  style: {},
+  tabIndex: 0,
+};
+
+export default function FileUploaderV2(props) {
+  const handleClick = ({ evt, index }) => props.onClick({ evt, index });
```

Comment:
Seems that we can do `const handleClick = props.onClick`?

Reply:
@asudoh I left that in case I needed to add some logic within the component itself I have now refactored it to directly call the `onClick` prop